app/scripts/controllers/monitoring.js

```diff
@@ -0,0 +1,311 @@
+
+    $scope.kindSelector = {
+      selected: {
+        kind: "All"
+      }
+    };
+    $scope.kinds = [
+      {
+        kind: "Pods"
+      },
+      {
+        kind: "Builds"
+      },
+      {
+        label: "Deployments",
+        kind: "ReplicationControllers"
+      },
+      {
+        kind: "All"
+      }
+    ];
+
+    $scope.logOptions = {
+      pods: {},
+      deployments: {},
+      builds: {}
+    };
+
+    $scope.logCanRun = {
+      pods: {},
+      deployments: {},
+      builds: {}
+    };
+
+    $scope.logEmpty = {
+      pods: {},
+      deployments: {},
+      builds: {}
+    };
+
+    $scope.filters = {
+      hideOlderResources: true,
+      text: ''
+    };
+
+    var ageFilteredBuilds, ageFilteredDeployments, ageFilteredPods;
+
+    // Check if the metrics service is available so we know when to show the tab.
+    MetricsService.isAvailable().then(function(available) {
+      $scope.metricsAvailable = available;
+    });
+
+    var orderByDate = $filter('orderObjectsByDate');
+
+    var filterExpressions = [];
+    var updateKeywords = function() {
+      if (!$scope.filters.text) {
+        filterExpressions = [];
+        return;
+      }
+
+      var keywords = _.uniq($scope.filters.text.split(/\s+/));
+      // Sort the longest keyword first.
+      keywords.sort(function(a, b){
+        return b.length - a.length;
+      });
+
+      // Convert the keyword to a case-insensitive regular expression for the filter.
+      filterExpressions = _.map(keywords, function(keyword) {
+        return new RegExp(_.escapeRegExp(keyword), "i");
+      });
+    };
+
+    // Only filter by keyword on certain fields.
+    var filterFields = [
+      'metadata.name'
+    ];
+
+    var filterForKeyword = function(resources) {
+      var filteredResources = resources;
+      if (!filterExpressions.length) {
+        return filteredResources;
+      }
+
+      // Find events that match all keywords.
+      angular.forEach(filterExpressions, function(regex) {
+        var matchesKeyword = function(event) {
+          var i;
+          for (i = 0; i < filterFields.length; i++) {
+            var value = _.get(event, filterFields[i]);
+            if (value && regex.test(value)) {
+              return true;
+            }
+          }
+
+          return false;
+        };
+
+        filteredResources = _.filter(filteredResources, matchesKeyword);
+      });
+      return filteredResources;
+    };
+
+    var filterAllResourcesForKeyword = function() {
+      $scope.filteredPods = filterForKeyword(ageFilteredPods);
+      $scope.filteredDeployments = filterForKeyword(ageFilteredDeployments);
+      $scope.filteredBuilds = filterForKeyword(ageFilteredBuilds);
+    };
+
+    var setPodLogVars = function(pod) {
+      $scope.logOptions.pods[pod.metadata.name] = {
+        container: pod.spec.containers[0].name
+      };
+      $scope.logCanRun.pods[pod.metadata.name] = !(_.includes(['New', 'Pending', 'Unknown'], pod.status.phase));
```

Comment:
Suggest putting the logCanRun logic into a service since it's also needed for the browse page controllers. Then it's in one place.